### PR TITLE
Improve front-end linting setup

### PR DESCRIPTION
**Pull request by callumacrae**
Some things (such as me just using completely random quotes all over the place) were getting through the linting setup, so I've made it a bit more strict.

I've also added prettier, which I'm a big fan of. I've ran it in this pull request so you can see the kind of changes it makes. Feel free to discuss.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,17 +1,21 @@
 module.exports = {
+  root: true,
   env: {
     browser: true,
     es2021: true,
   },
   extends: [
+    "eslint:recommended",
     "airbnb-base",
-    "plugin:cypress/recommended",
```
Comment (ethanmills) [inline .eslintrc.js:8]: Why remove this?

Comment (callumacrae) [reply .eslintrc.js:8]: I've moved it into the eslintrc in the cypress directory

File: spec/cypress/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  extends: ["plugin:cypress/recommended"],
+  rules: {
+    "import/no-commonjs": ["off"],
+  },
+};
```
Comment (ethanmills) [inline spec/cypress/.eslintrc.js:6]: Does this "stack" with the main file? Do those rules apply too?

Comment (callumacrae) [reply spec/cypress/.eslintrc.js:6]: yep! eslint looks upwards until it finds a `root: true` and merges everything it finds

Comment (ethanmills) [reply spec/cypress/.eslintrc.js:6]: Nice!

**Review by kicferk1 — APPROVED** [review]
Just to make sure, this is just adding prettier and result of running it, right?

**Comment by callumacrae** [comment]
> Just to make sure, this is just adding prettier and result of running it, right?

yep

**Review by ethanmills — APPROVED** [review]

**Review by rupollbabu — APPROVED** [review]

**Pull request merged**